# Bepaling van warmtecapaciteit van een onbekend materiaal


# Introductie

Onbekende materialen kunnen geïdentificeerd worden door hun eigenschappen te meten. Een van deze eigenschappen is de warmtecapaciteit. In dit practicum gaan we de warmtecapaciteit van een onbekend materiaal bepalen door middel van een calorimeter experiment. Daarbij wordt een bepaalde massa van het materiaal naar een bekende temperatuur gebracht waarna het in een bekende hoeveelheid water met bekende temperatuur wordt geplaatst. Door de temperatuur van het water te meten na het mengen kan de warmtecapaciteit van het onbekende materiaal worden berekend.

# Theorie

De soortelijke warmte $c$ van een materiaal is gedefinieerd als de hoeveelheid warmte $Q$ die nodig is om de temperatuur $T$ van een kilogram van het materiaal met één graad Celsius (of één Kelvin) te verhogen:

$$
    c = \frac{Q}{m \Delta T}
$$ (eq:heat_capacity)

Waarbij $Q$ de hoeveelheid warmte in Joules is, $m$ de massa in kilogram is en $\Delta T$ de verandering in temperatuur is. Gegeven de wet van Black, die stelt dat de totale hoeveelheid warmte in een geïsoleerd systeem constant blijft, kunnen we de warmte die het onbekende materiaal verliest gelijkstellen aan de warmte die het water opneemt:

$$
    Q_{materiaal} = -Q_{water} 
$$ (eq:black)

wanneer we de massa's en de begintemperaturen van beide systemen kennen, maar slechts een van de twee soortelijke warmtes, kunnen we de onbekende soortelijke warmte berekenen. We combineren vergelijkingen [](#eq:heat_capacity) en [](#eq:black) om de volgende vergelijking te krijgen:

$$
    T_e = \frac{c_w m_w T_{w,b}+c_m m_m T_{m,b}}{c_w m_w + c_m m_m}
$$ (eq:combined)

Waarbij de subscripts $b$ en $e$ respectievelijk staan voor begintoestand en eindtoestand, $w$ voor water en $m$ voor het onbekende materiaal.

Bij metingen aan verschillende massa's van het onbekende materiaal en vervolgens een least square fit aan bovenstaande vergelijking kunnen we een precieze waarde voor de soortelijke warmte van het onbekende materiaal bepalen. Dat is, wanneer de warmtecapaciteit van bijvoorbeeld de beker te verwaarlozen is.


# Methode en materialen

## Ontwerp
De bovenstaande theorie wordt gebruikt om de soortelijke warmte van een onbekend materiaal te bepalen. Het experiment bestaat uit het verwarmen van verschillende massa's van het onbekende materiaal tot een bekende temperatuur, waarna het in een bekende hoeveelheid water met bekende temperatuur wordt geplaats. Door de temperatuur van het water te meten na het mengen kan de warmtecapaciteit van het onbekende materiaal worden berekend. Om de tijd voor het meten van meerdere materialen te reduceren, worden de data van de verschillende groepen in het lokaal samengevoegd. Van tevoren is afgesproken welke massa's door welke groep worden gemeten, en hoeveel water er gebruikt wordt.

## Materialen
Hieronder staat de lijst van benodigde materialen bij deze proef:
- Calorimeter
- Thermometer of temperatuursensor
- Verwarmingsbron 
- Diverse massablokjes van onbekend materiaal
- Weegschaal
- Water
- Maatcilinder of maatbeker


```{figure} figures/c_onbekend_setup.jpg
:width: 70%
:label: fig_c_onbekend_setup

Een schematische weergave van de opstelling
```


## Procedure
Bespreek wie welke massa's van het onbekende materiaal gaat meten.
Bespreek ook hoeveel water er gebruikt gaat worden.
Bepaal de begintemperaturen.
Hevel het aantal afgesproken massa's in de maatbeker. 
Roer voorzicht zodat de temperatuur homogeen is.
Noteer de hoogste gemeten temperatuur, dit is $T_e$.
Wissel de metingen uit met de andere groepen en voer de data-analyse uit.

```{note}
Hieronder staat een voorbeeld hoe je een grafiek moet opslaan en vervolgens in je document kunt oproepen.
Het voorbeeld kun je natuurlijk straks verwijderen.
```

# Resultaten


In [24]:
# Sla figuren op met  
# 
# plt.savefig("Figuren/naam.png", dpi=450)
# waarbij naam vervangen wordt door de bestandsnaam. 
# Onderstaande voorbeeld code en output grafiek 
import numpy as np
import matplotlib.pyplot as plt
from scipy.optimize import curve_fit

#constants
m_bakkje=173.16
m_bakkje_met_water= 456.4
m_w = m_bakkje_met_water - m_bakkje
m_m=100
c_w=4.18e3

#data from groups
m_w= np.array([307.3,218,229.1,137.5,331.08,283.24,296.8])
m_b=np.array([101.8,100,100.6,50.2,49.95,100,51.1])
T_wb = np.array([20,20.3,20.2,21.2,22.5,22.5,20.6])
T_mb = np.array([69,67,69,67,67,67,69])
dT_w = np.array([1.6,2.3,2.6,1.9,0.7,0.7,1])
T_e =dT_w+T_wb 

m_w = m_w/1000
m_m=m_b/1000

cm = c_w*m_w*(T_wb-T_e)/(m_m*(T_e-T_mb))
mu_cm= np.mean(cm)
print(mu_cm)
print(cm)

y= c_w*m_w*(T_wb-T_e)
x = m_m*(T_e-T_mb)

def funcCm(x,c_m):
    return c_m*x

cm_val, cm_cov = curve_fit(funcCm, x,y,p0=(480))

c_m= cm_val[0]
sigma_cm = np.sqrt(cm_cov[0, 0])
print(cm_val)
print(sigma_cm)

# Bij cm[5] wijkt de soortelijk weerstand heel erg af van de andere waardes. 
# Daarom moet deze meting misschien geschrapt worden. Dit is natuurlijk niet 
# zeker maar toch worden de berekeningen een keer zonder deze meting gedaan.

#meting met elke ..[5] verwijderd. 

m_w2  = np.delete(m_w, 5)
m_m2  = np.delete(m_m, 5)
T_wb2 = np.delete(T_wb, 5)
T_mb2 = np.delete(T_mb, 5)
T_e2  = np.delete(T_e, 5)

y2= c_w*m_w2*(T_wb2-T_e2)
x2 = m_m2*(T_e2-T_mb2)

def funcCm2(x2,c_m):
    return c_m*x2

cm_val2, cm_cov2 = curve_fit(funcCm2, x2,y2,p0=(480))

c_m2= cm_val2[0]
sigma_cm2 = np.sqrt(cm_cov2[0, 0])
print(cm_val2)

print('De gevonden soortelijke warmte is {:.2f} ± {:.2f} J/(kg·K)'.format(c_m,sigma_cm))
print('De gevonden soortelijke warmte zonder de afwijkende meting is {:.2f} ± {:.2f}J/(kg·K)'.format(c_m2,sigma_cm2))
print('het onbekende materiaal zou ijzer/staal kunnen zijn, want dit materiaal heeft een soortelijke weerstand van ongeveer 450 J/(kg·K) ')

439.05846160618285
[425.92458117 472.03873874 535.71617912 495.52360036 442.78985835
 189.21466667 512.20160684]
[422.54559149]
49.243466229641825
[478.42370451]
De gevonden soortelijke warmte is 422.55 ± 49.24 J/(kg·K)
De gevonden soortelijke warmte zonder de afwijkende meting is 478.42 ± 19.36J/(kg·K)
het onbekende materiaal zou ijzer/staal kunnen zijn, want dit materiaal heeft een soortelijke weerstand van ongeveer 450 J/(kg·K) 


# Discussie en conclusie

Hier een korte discussie en conclusie over de resultaten van het experiment en de implicaties daarvan.
De gevonden soortelijke warmte is 422.55 ± 49.24 J/(kg·K)
De gevonden soortelijke warmte zonder de afwijkende meting is 478.42 ± 19.36J/(kg·K)
het onbekende materiaal zou ijzer/staal kunnen zijn, want dit materiaal heeft een soortelijke weerstand van ongeveer 450 J/(kg·K) 

Voor dit onderzoek zijn de metingen slechts zeven keer herhaald. De onzekerheid van de berekende soortelijke weerstand met deze metingen is dan ook meer dan 10 procent van de verkregen waarde. Uit de zesde meting kwam een soortelijke weerstand van 189 J/(kg·K), terwijl de ander metingen allemaal tussen de 425 en 535 J/(kg·K) lagen. Dit zou kunnen wijzen op een mislukte meting, daarom is de soortelijke weerstand nog een keer berekend zonder deze meting. Dit verkleint de onzekerheid. Metingen mogen niet zomaar verwijdert worden omdat het beter uitkomt, maar in dit geval week meting zes zo sygnificant af van de ander metingen dat deze als strijdig is beschouwd. Meting 6 wijkt namelijk ruim meer dan 2*sigma af van de ander metingen. Ook na het verwijderen van meting zes is de meetonzekerheid nog vrij groot. Dit zal hoogstwaarschijnelijk komen door onnauwkeurige temperatuurmetingen en warmteverlies aan de omgeving. 